src/extension/features/budget/fund-half/index.ts

```diff
@@ -0,0 +1,160 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { getBudgetService, getEntityManager } from 'toolkit/extension/utils/ynab';
+import { getEmberView } from 'toolkit/extension/utils/ember';
+
+// The concept here is that for odd monthly Target amounts there is a low half and a high half.
+// Example:  For $65.05 the low half is $32.52 (x2 = $65.04).  The high half is $32.53 (x2 - $65.06).
+enum HalfType {
+  Exact = 'Exact',
+  Low = 'Low',
+  High = 'High',
+}
+
+interface BudgetCategoryAmount {
+  budgetCategoryId: string;
+  amount: number;
+}
+
+export class FundHalf extends Feature {
+  private lowHalf: BudgetCategoryAmount[] = [];
+  private highHalf: BudgetCategoryAmount[] = [];
+
+  shouldInvoke() {
+    return true;
+  }
+
+  halfFundButton(text: string, amount: number, halfType: HalfType) {
+    const formattedAmount = formatCurrency(amount, true);
+    return $(`
+    <button class="budget-inspector-button fund-half" type="button" data-half-type="${HalfType[halfType]}">
+        <div>${text}</div>
+        <div>
+            <strong class="user-data" title="${formattedAmount}">
+                <span class="user-data currency positive">
+                    <bdi>$</bdi>${formattedAmount}</span>
+            </strong>
+        </div>
+    </button>`);
+  }
+
+  observe(changedNodes: Set<string>) {
+    if (!this.shouldInvoke()) return;
+
+    if (changedNodes.has('budget-inspector-button')) {
+      this.updateDOM();
+    }
+  }
+
+  updateDOM() {
+    if (!$('.budget-inspector').length) return;
+    if (!$('.inspector-quick-budget').length) return;
+    if ($('.budget-inspector-button.fund-half').length) return;
+
+    const selectedBudgetCategories = this._getSelectedBudgetCategories();
+    if (selectedBudgetCategories != null) {
+      const underfundedButtonText = $($('.inspector-quick-budget .option-groups > div')[0]).find(
```

Comment:
I think you can do something like 

```js
const underfundedButtonText = $('.inspector-quick-budget .option-groups > div').eq(0).find("button div:contains('Underfunded')");
```

And avoid double call to jQuery

Reply:
@OlegWock Thanks, fixed!  I haven't done jQuery in about 7 years so I appreciate the refresher and guidance.